sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/ResolveDefaultColumnsUtil.scala

```diff
@@ -52,6 +52,17 @@ object ResolveDefaultColumns extends QueryErrorsBase
   // CURRENT_DEFAULT_COLUMN_METADATA.
   val CURRENT_DEFAULT_COLUMN_NAME = "DEFAULT"
 
+  var defaultColumnAnalyzer: Analyzer = DefaultColumnAnalyzer
+  var defaultColumnOptimizer: Optimizer = DefaultColumnOptimizer
+
+  /**
+   * Visible for testing
+   */
+  def setAnalyzerAndOptimizer(analyzer: Analyzer, optimizer: Optimizer): Unit = {
```

Comment:
Its hard to reproduce the issue in unit test, so I end up mocking these members to verify that the catalogs are not called.